src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -217,11 +227,8 @@ namespace winrt::TerminalApp::implementation
         if (_currentMode == CommandPaletteMode::TabSwitchMode)
         {
             const auto selectedCommand = _filteredActionsView().SelectedItem();
-            if (const auto filteredCommand = selectedCommand.try_as<winrt::TerminalApp::FilteredCommand>())
-            {
-                const auto& actionAndArgs = filteredCommand.Command().Action();
-                _dispatch.DoAction(actionAndArgs);
-            }
+            const auto filteredCommand{ selectedCommand.try_as<winrt::TerminalApp::FilteredCommand>() };
+            _switchToTab(filteredCommand);
```

Comment:
@zadjii-msft -  are you addressing the readability or the runtime flow? From the code readability it is still far from perfect, but from the runtime perspective it will work happen only for tabs (as the function is invoked only in TabSwitch mode).